docs/a2a/fasta2a.md

```diff
@@ -0,0 +1,64 @@
+## FastA2A
+
+!!! warning
+    This page is WIP.
+
+To make it easier to implement A2A servers, we've implemented FastA2A,
+a library built on top of Starlette and Pydantic to bring A2A to Python.
+
+You can use FastA2A with any agentic framework (it's not exclusive
+to PydanticAI). That said, we have a convenience method to create an
+A2A server from an existing PydanticAI agent.
+
+```python {title="a2a.py}
+from pydantic_ai.agent import Agent
+
+agent = Agent('anthropic:claude-3-5-sonnet-latest', name='My A2A Agent')
+app = agent.to_a2a()
+```
+
+Then you can run the server with:
+
+```bash
+uvicorn a2a:app --host 0.0.0.0 --port 8000
+```
+
+### Design
+
+We built `FastA2A` with the following design in mind:
+
+```mermaid
+flowchart TB
+    Server["HTTP Server"] <--> |Sends Requests/<br>Receives Results| TM
+
+    subgraph CC[Core Components]
+        direction RL
+        TM["TaskManager<br>(coordinates)"] --> |Schedules Tasks| TE
+        TM <--> Storage
+        TE["Worker<br>(queues & exec.)"] <--> Storage["Storage<br>(persistence)"]
+        TE --> |Delegates Execution| Runner
+    end
+
+    Runner["Runner<br>(implementation)"]
+```
+
+You can bring your own `Storage`, `Worker` and `Runner`.
```

Comment:
I'd provide examples here of the technologies you have in mind (e.g. celery? something else?). Just to help make it clear why one might _want_ to bring their own.